### PR TITLE
fixes error creating a new database due to Malformed JSON error for empty Secure extras field

### LINKED ISSUE #8607: Malformed JSON in the extras field: SyntaxError: Unexpected end of JSON input
While adding a new database connection (tried for Postgres and Redshift both), clicking on '**Test Connection**' gives  `Malformed JSON in the extras field: SyntaxError: Unexpected end of JSON input` error.
The **Extra** field has 
```
{
    "metadata_params": {},
    "engine_params": {},
    "metadata_cache_timeout": {},
    "schemas_allowed_for_csv_upload": []
}
```
And the **Secure Extra** field is empty. I also tried removing the JSON in the **Extra** field altogether but I still get this error. 

The connection strings, I've tried are:
`redshift+psycopg2://user:pass@host/dbname` for Redshift and `postgresql://user:pass@host:5432/dbname` for Postgres 

### Expected results
The connection should be tested successfully.

### Actual results
Clicking on '**Test Connection**' gives  `Malformed JSON in the extras field: SyntaxError: Unexpected end of JSON input` error.

#### Screenshots
Screenshot 2019-11-20 at 15 19 36

#### How to reproduce the bug

1. Go to Sources > Databases
2. Click on 'Add a new record' (Plus button)
3. Keeping all the field as it is, just add the database name and database URI
4. Click on 'Test Connection' to see the error.

### Environment

(please complete the following information):

- superset version: `"0.999.0dev",`
- python version: `Python 3.6.7`
- node.js version: `v13.1.0`
- npm version: `6.12.1`

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I have copied the default JSON in the extras field and checked its validity, and it is a valid JSON.

### LINKED ISSUE #8628: Cannot add new database connection due to Malformed JSON in the extras field error
I get a _Malformed JSON in the extras field: SyntaxError: Unexpected end of JSON input_ error when I try to test the connection of a new database. 

### Expected results

Successful connection test indicated by Javascript alert modal that says _Seems OK!_

### Actual results

Javascript alert modal that says _Malformed JSON in the extras field: SyntaxError: Unexpected end of JSON input_.

#### Screenshots

Error I get when I test the connection (note: disregard `hi` at beginning of alert - I was experimenting with a fix):

Screen Shot 2019-11-21 at 4 17 57 PM

`Secure extra` field is blank:

Screen Shot 2019-11-21 at 4 18 05 PM

#### How to reproduce the bug

```
git clone https://github.com/apache/incubator-superset/
cd incubator-superset/contrib/docker
docker-compose run --rm superset ./docker-init.sh
docker-compose up
```

Then...
- open the UI
- go to add a database
- configure the database connection, making sure to keep the `Secure extra` field blank, and click "Test Connection"

### Environment

Superset running in Docker (development mode)
Superset cloned from master branch (commit: b7a508d5a7a690ef56180a1b2e7ec151c6572d52)

### Checklist

- [ X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X ] I have reproduced the issue with at least the latest released version of superset.
- [ X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

I believe this bug was introduced in [this PR](https://github.com/apache/incubator-superset/commit/d70e0fc3593218cc562eeb68ef4311e41d08aeb0#diff-8d6d8d888724e94b5b4f2f3e46cb2fc7).

Workaround: add empty object (`{}`) to `Secure extras` field to successfully test new database connection.

Screen Shot 2019-11-21 at 4 18 12 PM
Screen Shot 2019-11-21 at 4 18 19 PM

### PR DESCRIPTION
**Pull request by maxmelnick**
### CATEGORY

Choose one

- [X] Bug Fix

### SUMMARY
Fixes #8628 and Fixes #8607 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

See #8628 for before screenshots.

After screenshots demonstrate I am able to have a blank `Secure extras` field and successfully test a new connection.

Screen Shot 2019-11-21 at 4 18 19 PM
Screen Shot 2019-11-21 at 4 18 05 PM

### TEST PLAN

Tested locally in docker.

```
git clone https://github.com/apache/incubator-superset/
cd incubator-superset/contrib/docker
docker-compose run --rm superset ./docker-init.sh
docker-compose up
# then tested adding a new database in UI and was successful
```

Note: I tested an empty `Secure extra` field, as well as explicitly setting `Secure extra` field = `{}` in UI to test case when someone provides valid JSON for `Secure extra` field.

### ADDITIONAL INFORMATION
- [ X] Has associated issue: Fixes #8628 and Fixes #8607 

### REVIEWERS
@dpgaspar

### CODE REVIEW

File: superset/templates/superset/models/database/macros.html [hunk 1]
```diff
@@ -40,7 +40,7 @@
           name: $('#database_name').val(),
           impersonate_user: $('#impersonate_user').is(':checked'),
           extras: JSON.parse($("#extra").val()),
-          encrypted_extra: JSON.parse($("#encrypted_extra").val()),
+          encrypted_extra: $("#encrypted_extra").val() ? JSON.parse($("#encrypted_extra").val()) : {},
```
Comment (mistercrunch) [inline superset/templates/superset/models/database/macros.html]: can we  `const encryptedExtra = $("#encrypted_extra").val();` above the try block first?

Comment (maxmelnick) [reply superset/templates/superset/models/database/macros.html]: @mistercrunch - good point. I just pushed that change.

**Comment by mistercrunch** [comment]
Do we not need to do the same for `#extra`, or is it because the default value of `{}` was only set for one of the two?

**Comment by gamebusterz** [comment]
> Do we not need to do the same for `#extra`, or is it because the default value of `{}` was only set for one of the two?

Could be, because the `#extra` field is what the error message was pointing to.

**Comment by dpgaspar** [comment]
It seems like it would be better to do the same for `extra` also, a user can simply clean that field and test the connection causing it to fail with the same error

**Comment by maxmelnick** [comment]
Sounds good. I added handling for an empty `extra` field as well.

**Review by dpgaspar — APPROVED** [review]
LGTM

**Pull request merged**